extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -438,7 +438,7 @@ export class AmpAnalytics extends AMP.BaseElement {
     request = this.expandTemplate_(request, trigger, event);
 
     // For consistency with amp-pixel we also expand any url replacements.
-    return urlReplacementsForDoc(this.win.document).expandAsync(request)
+    return urlReplacementsForDoc(this.getAmpDoc()).expandAsync(request)
```

Comment:
@dvoytenko using `this.win.document` I get errors like:
```Error: No ampdoc found for [object HTMLDocument]```

Is this supposed to work? 

What seems to be working is to pass `this.element` as we discussed.